crates/sys/src/lib.rs

```diff
@@ -48,21 +48,79 @@ mod bindings;
 /// ```
 /// flipperzero_sys::crash!("Hello world!");
 /// ```
+///
+/// Crash the system with default *"Fatal Error"* message:
+///
+/// ```
+/// flipperzero_sys::crash!();
+/// ```
 #[macro_export]
 macro_rules! crash {
-    ($msg:expr $(,)?) => {{
-        const MESSAGE: *const ::core::primitive::i8 =
-            match ::core::ffi::CStr::from_bytes_with_nul(
-                ::core::concat!($msg, "\0").as_bytes(),
-            ) {
-                Ok(cstr) => cstr.as_ptr(),
-                Err(error) => panic!("message contains NULs"),
+    ($($msg:expr)? $(,)?) => {{
+        // Optional message
+        let mut message: *const i8 = ::core::ptr::null();
```

Comment:
I am  not a  big fan  of this maybe-redundant initialization, so I would  prefer this to (still)  be  a  `($msg:expr $(,)?)` branch with an empty branch `()` delegating to it
Also note that this  would fix  the  (currently added) problem of `crash(,)` being a valid expression.

Reply:
Sounds good. I agree that's a better approach.

Reply:
Since the crash handler specifically requires a pointer, I added a `__crash/halt_implementation!` macro that takes the pointer from the public `crash/halt!` macros rather than resorting to pattern-matching trickery.